src/graph/executor/algo/SubgraphExecutor.cpp

```diff
@@ -21,55 +21,69 @@ folly::Future<Status> SubgraphExecutor::execute() {
   DCHECK(currentStepVal.isInt());
   auto currentStep = currentStepVal.getInt();
   VLOG(1) << "Current Step is: " << currentStep << " Total Steps is: " << steps;
-
-  if (currentStep == steps) {
-    oneMoreStep();
-    return finish(ResultBuilder().value(Value(std::move(ds))).build());
-  }
+  auto resultVar = subgraph->resultVar();
 
   VLOG(1) << "input: " << subgraph->inputVar() << " output: " << node()->outputVar();
   auto iter = ectx_->getResult(subgraph->inputVar()).iter();
-  DCHECK(iter && iter->isGetNeighborsIter());
+
+  ResultBuilder builder;
+  builder.value(iter->valuePtr());
+
+  std::unordered_map<Value, int64_t> currentVids;
+  currentVids.reserve(iter->size());
```

Comment:
assign `iter->size()` to a local variable to avoid to repeat to compute the size.

Reply:
ok